modules/web-console/web-console-common/src/main/java/org/apache/ignite/console/event/Event.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2019 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.console.event;
+
+import org.springframework.context.ApplicationEvent;
+
+/**
+ * Generic application event with payload
+ */
+public class Event<T> extends ApplicationEvent {
```

Comment:
I think we should rename this event to AccountEvent and T should extend Account.
Without this all non account listeners will be notified about account event.

Reply:
I've implement ResolvableTypeProvider interface and resolve issue with generic type erasure.